Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/FitGaussianPeaks.py

```diff
@@ -285,6 +285,9 @@ def estimate_single_parameters(self, xvals, yvals, centre, win_size):
             rside = 5 + lside
         x_range = xvals[lside:rside]
         y_range = yvals[lside:rside]
+        if len(x_range) < 5:
```

Comment:
why 5?

Reply:
There are 5 parameters to be fitted which need a minimum of 5 points

Reply:
That would still be a bad fit https://stats.stackexchange.com/questions/37833/minimal-number-of-points-for-a-linear-regression . Would it be worth making it a parameter so users can say that it needs 20 data points? Then we just validate that the value is at least some minimal value.

Reply:
The window size which determines the number of data points used is determined by EstimatedPeakSigma argument given by users

Reply:
so why is this five hard coded if it is a user defined value?

Reply:
Code checks if range used is less than 5 and if it is, it tries to correct it so it uses at least 5 datapoints and if it cannot correct it  it skips the  peak and raises a warning

Reply:
you can validate the user value is 5 or more instead. This would give the user quicker feedback

Reply:
I suppose the best way might be for users to specify the window as EstimatedPeakSigma is used for other things as well

Reply:
yes please

Reply:
May I just query whether this `< 5` check was left intentionally or was to be removed in favour of the `EstimatedPeakSigma`? I can't quite tell what the fainl intention was.

If we are keeping the check can we replace the `5` with a named module level constant like `MIN_NUM_DATA_PTS` then it would be a little easier to read off the intent of the variable:

```python
if len(x_range) < MIN_NUM_DATA_PTS:
```